scripts/update-snapshots.mjs

```diff
@@ -4,87 +4,121 @@ import { exec } from 'child_process';
 import fs from 'fs';
 import path from 'path';
 
-// Read and parse package.json
-const packageJsonPath = path.join(process.cwd(), 'package.json');
-const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
-const dependencies = packageJson.dependencies || {};
-const devDependencies = packageJson.devDependencies || {};
-const allDependencies = Object.assign({}, dependencies, devDependencies);
-
-// Snapshot version as the first command-line argument
+const workingDir = process.cwd();
+
+// Function to check if the script is run from the correct directory
+function verifyRootDirectory(expectedPackageName) {
+    const packageJsonPath = path.join(workingDir, 'package.json');
+    if (!fs.existsSync(packageJsonPath)) {
+        console.error("Error: package.json not found. Please run this script from the root directory of your project.");
+        process.exit(1);
+    }
+
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+    if (packageJson.name !== expectedPackageName) {
+        console.error(`Error: This script should be run from the root directory of '${expectedPackageName}'.`);
+        process.exit(1);
+    }
+}
+
+verifyRootDirectory('pie-aperture'); // Ensure the script is run from the root directory of 'pie-aperture'
+
+const subProjects = ['nuxt-app', 'vanilla-app', 'nextjs-app'];
+
+function readDependencies(filePath) {
+    if (fs.existsSync(filePath)) {
+        const packageJson = JSON.parse(fs.readFileSync(filePath, 'utf8'));
+        return { ...packageJson.dependencies, ...packageJson.devDependencies };
+    }
+    return {};
+}
+
+function collectDependencies() {
+    const allDependencies = {};
+    subProjects.forEach(subProject => {
+        const packageJsonPath = path.join(workingDir, subProject, 'package.json');
+        const deps = readDependencies(packageJsonPath);
+        for (const [key, value] of Object.entries(deps)) {
+            if (key.startsWith('@justeattakeaway/')) {
+                allDependencies[key] = value;
+            }
+        }
+    });
+    return allDependencies;
+}
+
 const snapshotVersion = process.argv[2];
 const packagesToUpdate = process.argv.slice(3);
+const versionTag = `0.0.0-snapshot-release-${snapshotVersion}`;
 
-// Filter for packages that are in the list of packages to update and start with @justeattakeaway/pie-
+if (!snapshotVersion || !/^\d{14}$/.test(snapshotVersion)) {
+    console.error("Error: Invalid or missing snapshot version. Please provide a snapshot version in the format YYYYMMDDHHMMSS (e.g., 20240416153654).");
+    process.exit(1);
+}
+
+if (!packagesToUpdate.length) {
+    console.error("Error: No packages specified for update.");
+    process.exit(1);
+}
+
+const allDependencies = collectDependencies();
 const basePackages = Object.keys(allDependencies).filter(name =>
-    packagesToUpdate.includes(name) && name.startsWith('@justeattakeaway/pie-')
+    packagesToUpdate.includes(name) && name.startsWith('@justeattakeaway/')
```

Comment:
I guess `allDependencies` at this stage already only includes keys that start with `@justeattakeaway/`?

I wonder if working with a `Set` object would be a little easier than an object, especially considering we don't really care about the old version numbers?

If `allDependencies` and `packagesToUpdate` were both Sets, you can do cool things like this:
```js
const basePackages = allPackages.intersection(packagesToUpdate);
```

Reply:
As discussed, `intersection` isn't available in NodeJS - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/intersection#browser_compatibility